features/roomlist/impl/src/main/kotlin/io/element/android/features/roomlist/impl/model/RoomListRoomSummary.kt

```diff
@@ -31,4 +32,5 @@ data class RoomListRoomSummary constructor(
     val lastMessage: CharSequence? = null,
     val avatarData: AvatarData = AvatarData(id, name, size = AvatarSize.RoomListItem),
     val isPlaceholder: Boolean = false,
+    val notificationMode: RoomNotificationMode? = null,
```

Comment:
Is there any semantic difference between `RoomNotificationMode.ALL_MESSAGES` and `null` here?
If not then it might be considered to make this val non-nullable and not expose the null case to the UI so that the resulting logic can be simpler.

Reply:
At the moment it represents the user defined room level notification setting. So if the user does not set one it is null. The semantics are quite in flux at the moment, but when they settle maybe we can change this to a non-matrix type as you suggested offline.